src/ja/kisslove/src/eu/kanade/tachiyomi/extension/ja/kisslove/KissLove.kt

```diff
@@ -70,13 +70,24 @@ class KissLove : FMReader("KissLove", "https://klz9.com", "ja") {
             ?.`val`()
             ?: throw Exception("Could not find chapter id")
 
-        val xhrUrl = "$baseUrl/app/manga/controllers/cont.listImg.php".toHttpUrl().newBuilder()
+        val xhrUrl = "$baseUrl/${generateRandomStr(30)}.iog".toHttpUrl().newBuilder()
             .addQueryParameter("cid", chapterId)
             .build()
 
         return GET(xhrUrl, headers)
     }
 
+    private fun generateRandomStr(length: Int): String {
+        val characters = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789"
+        val charactersLength = characters.length
+        var result = ""
+        for (i in 0..length) {
```

Comment:
This would add an extra character.

Also, this can mostly be replaced with:
```kt
return (1..length).map { characters.random() }.joinToString("")
```

Reply:
Thank you.
It has been changed.